### ISSUE TITLE
Cannot override `clusters` when run as systemd service

### ISSUE THREAD

**Issue by faximan** [buildbarn/bb-clientd#19]
Hey!

As I'm not familiar with jsonnet I am probably just holding this wrong... I have figured out that I need to override the `clusters` variable in `bb_clientd.jsonnet` to include my RBE endpoint.

This works when run from source, but when run as a [systemd service](https://github.com/buildbarn/bb-clientd?tab=readme-ov-file#-as-a-systemd-service) that does not give me access to the config directly, I am not sure what I need to put into `~/.config/bb_clientd.jsonnet`.

Thanks in advance,

**Comment by EdSchouten** [comment]
Yeah, that's a bit annoying. Right now you need to copy the entire default configuration to `~/.config/bb_clientd.jsonnet` and adjust that. Ideally we'd make it so that you could override just that part:

```jsonnet
local defaults = import 'bb_clientd_defaults.jsonnet';
defaults {
  clusters:: ['hostname-a', 'hostname-b'],
}
```

Unfortunately, that's not supported (yet). Patches are welcome!

**Comment by moroten** [comment]
Solved by https://github.com/buildbarn/bb-clientd/pull/23.

**Comment by adam-azarchs** [comment]
We've got some additional changes our (internal) fork to further enhance extendibility like that, which I should probably upstream, particularly making `cacheDirectory` also be a member rather than a local so that it can be overridden.  Basically you do that and you can have your config file look something like
```jsonnet
(import '/usr/lib/bb_clientd/bb_clientd.jsonnet') + {
  cacheDirectory:: '/mnt/bazelbuild/bb_clientd',
  casBlocksSizeBytes:: 100 * 1024 * 1024 * 1024,
  filePoolSizeBytes:: 100 * 1024 * 1024 * 1024,
  global+: {
    diagnosticsHttpServer: {
      httpServers: [{
        listenAddresses: ['127.0.0.1:4321'],
        authenticationPolicy: { allow: {} },
      }],
      enablePrometheus: true,
    },
  },
}
```
